---
source: crates/biome_js_analyze/tests/spec_tests.rs
expression: checksConditionalsInvalid.ts
---
# Input
```ts
const promise = Promise.resolve('value');

if (promise) {
  // Do something
}

const val = promise ? 123 : 456;

// FIXME: Not yet detected
[1, 2, 3].filter(() => promise);

while (promise) {
  // Do something
}

```

# Diagnostics
```
checksConditionalsInvalid.ts:3:5 lint/nursery/noMisusedPromises ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  i A Promise was found where a conditional was expected.
  
    1 │ const promise = Promise.resolve('value');
    2 │ 
  > 3 │ if (promise) {
      │     ^^^^^^^
    4 │   // Do something
    5 │ }
  
  i A Promise is always truthy, so this is most likely a mistake.
  
  i You may have intended to `await` the Promise instead.
  

```

```
checksConditionalsInvalid.ts:7:13 lint/nursery/noMisusedPromises ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  i A Promise was found where a conditional was expected.
  
    5 │ }
    6 │ 
  > 7 │ const val = promise ? 123 : 456;
      │             ^^^^^^^
    8 │ 
    9 │ // FIXME: Not yet detected
  
  i A Promise is always truthy, so this is most likely a mistake.
  
  i You may have intended to `await` the Promise instead.
  

```

```
checksConditionalsInvalid.ts:12:8 lint/nursery/noMisusedPromises ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  i A Promise was found where a conditional was expected.
  
    10 │ [1, 2, 3].filter(() => promise);
    11 │ 
  > 12 │ while (promise) {
       │        ^^^^^^^
    13 │   // Do something
    14 │ }
  
  i A Promise is always truthy, so this is most likely a mistake.
  
  i You may have intended to `await` the Promise instead.
  

```
